src/nested_ragged_tensors/ragged_numpy.py

```
@@ -989,124 +1217,172 @@
+                try:
+                    L = bounds.get_shape()[0]
+                except Exception:
+                    L = len(bounds)
 
-                        vals_start = offset
-                        vals_end = B[-1] + offset
+                if st_i != 0:
+                    if st_i <= L:
+                        new_st_i = bounds[st_i - 1]
+                    else:
+                        new_st_i = bounds[-1]
```

Ruff

Use ternary operator `new_st_i = bounds[st_i - 1] if st_i <= L else bounds[-1]` instead of `if`-`else`-block

Replace `if`-`else`-block with `new_st_i = bounds[st_i - 1] if st_i <= L else bounds[-1]`

(SIM108)